hax/hax/motr/delivery.py

```diff
@@ -101,6 +79,8 @@ def wait_for_all(self,
                         confirmed_msgs)
                 del self.waiting_clients[promise]
                 promise.exclude_ids(confirmed_msgs)
+                if option.lower() == "any":
```

Comment:
Let's avoid this and keep the 2 explicitly named functions but common out the content as mentioned above.